lib/src/main/java/io/ably/lib/realtime/Presence.java

```diff
@@ -26,37 +26,49 @@
 	 ************************************/
 
 	/**
-	 * Get the presence state for this Channel. Implicitly attaches the
-	 * Channel. However, if the channel is in or moves to the FAILED
-	 * state before the operation succeeds, it will result in an error
-	 * @return: the current present members.
+	 * String parameter names to get() call with Param[] as an argument
+	 */
+	public final static String GET_WAITFORSYNC = "waitForSync";
+	public final static String GET_CLIENTID = "clientId";
+	public final static String GET_CONNECTIONID = "connectionId";
+
+	/**
+	 * Get the presence state for this channel. Take Param[] array as an argument
+	 * @param params
+	 * @return
 	 * @throws AblyException
+	 * @throws InterruptedException
 	 */
-	public synchronized PresenceMessage[] get() throws AblyException {
+	public synchronized PresenceMessage[] get(Param... params) throws InterruptedException, AblyException {
```

Comment:
It might cause another problem. It is an error in Java to have `catch` block for the exception never thrown in the `try` block. So if we move to `InterruptedException` integrated into `AblyException` we need to modify methods `Presence.get(boolean wait)` and `Presence.get(String clientId, boolean wait)` too so they'll stop throwing `InterruptedException`. And this in turn will break code that uses old `Presence.get()` stuff.